stl/inc/tuple

```diff
@@ -667,11 +676,23 @@ public:
         return _Myfirst._Val == _Right._Myfirst._Val && _Mybase::_Equals(_Right._Get_rest());
     }
 
+#ifdef __cpp_lib_concepts
+    template <class _Other_first, class... _Other_rest>
+    constexpr common_comparison_category_t<_Synth_three_way_result<_This, _Other_first>,
+        decltype(declval<const tuple<_Rest...>>()._Three_way_compare(declval<const tuple<_Other_rest...>>()))>

```

Comment:
I believe that I do need to handle `_Other_first` and `_Other_rest` separately as we don't save the parameter pack of all the types of the left hand tuple.

Your suggestion breaks when comparing two tuples which are legitimately comparable but don't have the same underlying type. e.g. `tuple<double, long, int>` and `tuple<long, int, double>`. If I followed your suggestions we would have `tuple<double, long>` and `tuple<long, int>` returning `std::strong_ordering`. It only works for `_Equals` because it always returns a `bool`.